packages/aws-cdk-lib/aws-apigateway/lib/lambda-api.ts

```diff
@@ -86,13 +87,13 @@ export class LambdaRestApi extends RestApi {
 }
 
 function addResourceThrows(): Resource {
-  throw new Error('Cannot call \'addResource\' on a proxying LambdaRestApi; set \'proxy\' to false');
+  throw new UnscopedValidationError('Cannot call \'addResource\' on a proxying LambdaRestApi; set \'proxy\' to false');
```

Comment:
these 3 methods are not public so we can change them. i think we should just add `scope: Construct` to the function parameters so these don't have to be unscoped

Reply:
This ends up getting a bit funky due to the way these are being used. We end up needing to use `this` and it just gets weird so I'm gonna keep these as is for now